oml/metrics/embeddings.py

```diff
@@ -147,12 +169,104 @@ def compute_metrics(self) -> TMetricsDict_ByLabels:  # type: ignore
                     distances=self.distance_matrix[mask],  # type: ignore
                     mask_gt=self.mask_gt[mask],  # type: ignore
                     mask_to_ignore=self.mask_to_ignore[mask],  # type: ignore
+                    reduce=False,
                     **args,  # type: ignore
                 )
 
-        self.metrics = metrics  # type: ignore
+        self.metrics_unreduced = metrics
+        self.metrics = reduce_metrics(metrics)  # type: ignore
+
+        return self.metrics
+
+    def get_worst_queries_ids(self, metric_name: str, topk: int) -> List[int]:
+        metric_values = flatten_dict(self.metrics_unreduced)[metric_name]  # type: ignore
```

Comment:
Do we need to use `flatten_dict` here? In reality this function is used only for tensorboard groupping of images)

Really not related to metric.

Reply:
I don't see any good alternative, maybe you can suggest something?

Reply:
I mean, we want to have consistent metric names and that's why I reuse `flatten_dict`, instead of creating something else.